prog/analysis.go

```diff
@@ -12,19 +12,79 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"strings"
 
 	"github.com/google/syzkaller/pkg/image"
 )
 
 type state struct {
-	target    *Target
-	ct        *ChoiceTable
-	corpus    []*Prog
-	files     map[string]bool
-	resources map[string][]*ResultArg
-	strings   map[string]bool
-	ma        *memAlloc
-	va        *vmaAlloc
+	target     *Target
+	ct         *ChoiceTable
+	corpus     []*Prog
+	files      map[string]bool
+	resources  map[string][]*ResultArg
+	strings    map[string]bool
+	ma         *memAlloc
+	va         *vmaAlloc
+	ioctlStack []string
+}
+
+func pushFieldToStack(s *state, fName string) {
```

Comment:
It would really help to have some detailed comment regarding these new methods - what they are for / what kind of data `ioctlStack` actually represents.